python/cudf/cudf/core/multiindex.py

```diff
@@ -438,7 +438,11 @@ def take(self, indices):
         if isinstance(indices, (Integral, Sequence)):
             indices = np.array(indices)
         elif isinstance(indices, Series):
-            indices = indices.data.mem
+            indices = (
+                indices.fillna(False).data.mem
+                if indices.null_count != 0
+                else indices.data.mem
+            )
```

Comment:
This is in the `take(...)` function which is essentially a gather. If we have nulls you could either argue that we should propagate a null from the input map to the output (which as far as I'm aware we don't currently do), or raise an Exception that nulls in the gather map aren't allowed.